sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,210 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Binary indicators for missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or np.NaN, optional (default=np.NaN)
+        The placeholder for the missing values. All occurrences of
+        ``missing_values`` will be imputed.
+
+    features : {'train' (default), 'all', array-like of int}
+        If "all", mask will represent all features.
+        If "train", mask will only represent features with missing values
+        during fit time.
+        If mask/indices, mask will only represent features in the
+        indices or mask.
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be a sparse matrix.
+        If False, the transformed ``X`` will be a numpy array.
+        If "auto", the transformed ``X`` will be of same type as input.
+
+    strict : boolean, optional (default=True)
+        If True, transform will raise an error when there are features with
+        missing values in transform but have no missing values in fit
```

Comment:
I'm not sure if ``strict`` is a good name for this. maybe ``error_on_unknown``? ``handle_unknown`` is what we have in OneHotEncoder.. maybe here it's more ``error_on_new``?